### PR TITLE
Creating Typed Predicates With KeyPaths

### PR DESCRIPTION
**Pull request by designatednerd**
I hate Stringly-typed code SO MUCH, and while `NSSortDescriptor` got a standard lib `keyPath:` initializer, `NSPredicate` didn't, and is the source of tons of awful stringly-typed code. 

With inspiration from @kishikawakatsumi's [Kuery]( https://github.com/kishikawakatsumi/Kuery) lib, I pulled a bunch of stuff around creating `NSPredicate`s with `KeyPath`s into our library, and then tested it like a crazy person. 

Notes:
- I also threw in a couple convenience methods on `NSManagedObject` and `NSManagedObjectContext`. If you think those belong in a separate `Sweet*` lib, let me know which one, and then i'll pull the categories here into the test bundle.
- Removed a deprecated method, so we'll do a major version bump on this.

### CODE REVIEW

**Comment by elland** [comment]
Nice! Always fun to get rid of strings! Looking forward to taking a closer look.

File: Sources/NSManagedObject+Sweetness.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//
+//  NSManagedObject+Sweetness.swift
+//  Tests
+//
+//  Created by Ellen Shapiro on 11/23/18.
+//
+
+import CoreData
```
Comment (3lvis) [inline Sources/NSManagedObject+Sweetness.swift]: Initially, we discussed that the scope of this project was going to be limited by using `import Foundation`, if another import was needed it was out of scope. For that case we also made [SweetCoreData](https://github.com/UseSweet/SweetCoreData).

**Comment by designatednerd** [comment]
@3lvis @Elland moved the core data bits to the test bundle (finally). Ready for a full look.

File: Sources/Array+Sweetness.swift [hunk 2]
```diff
@@ -1,9 +0,0 @@
-import Foundation
```
Comment (elland) [inline Sources/Array+Sweetness.swift:1]: We could probably use a quick PR just removing this 👍

Comment (designatednerd) [reply Sources/Array+Sweetness.swift:1]: I figured I should lump this in with a major version bump

File: Sources/NSPredicate+Sweetness.swift [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            return NSPredicate(
+                format: "%K BETWEEN %@",
+                argumentArray: [
+                    keyPath.toNSString,
+                    [ firstValue._object, secondValue._object ]
+                ])
+        }
+    }
+    
+    private static func fakeBetween<Root, Property: ComparableProperty>(_ firstValue: Property,
```
Comment (elland) [inline Sources/NSPredicate+Sweetness.swift]: Could use a clearer name and documentation 👍🏽

Comment (designatednerd) [reply Sources/NSPredicate+Sweetness.swift]: I changed the name in a way that I think is self-documenting - let me know if you think it still needs more info

File: Sources/Array+Sweetness.swift (hunk 2 above)
Comment (elland) [reply Sources/Array+Sweetness.swift:1]: 👍🏽

**Comment by designatednerd** [comment]
@elland any further thoughts? 

@3lvis @yuliaveres how 'bout you?

**Comment by elland** [comment]
@designatednerd busy week, reviewing slow as time allows 🙈

**Comment by designatednerd** [comment]
@3lvis @Elland @yuliaveres any further feedback? If I don't hear anything back before lunch tomorrow (15 January) I'm gonna merge this.

**Comment by elland** [comment]
Sorry I missed! I'll take a last look tomorrow for sure. 💪🏽

**Review by elland — APPROVED** [review]
🚀

**Pull request merged**